src/librbd/LibrbdWriteback.cc

```diff
@@ -240,10 +242,10 @@ namespace librbd {
       if (new_journal_tid != 0) {
         // ensure new journal event is safely committed to disk before
         // committing old event
-        m_ictx->journal->flush_event(
-          new_journal_tid, new C_CommitIOEventExtent(m_ictx,
-                                                     original_journal_tid,
-                                                     it->first, it->second));
+        Context *ctx = new C_CommitIOEventExtent(m_ictx, original_journal_tid,
+                                                 it->first, it->second);
+        ctx = new C_RefRequestCallback<Journal<>>(m_ictx->journal, ctx);
```

Comment:
Wouldn't you need to protect the ```else``` clause as well?

Reply:
The `else` part isn't asynchronous, right?

Reply:
Ah — never mind